test/Ocelot.AcceptanceTests/PollyQoSTests.cs

```diff
@@ -1,276 +1,311 @@
-                .And(x => _steps.WhenIGetUrlOnTheApiGateway("/"))
-                .And(x => _steps.ThenTheStatusCodeShouldBe(HttpStatusCode.ServiceUnavailable))
-                .And(x => _steps.WhenIGetUrlOnTheApiGateway("/"))
-                .And(x => _steps.ThenTheStatusCodeShouldBe(HttpStatusCode.ServiceUnavailable))
-                .And(x => GivenIWaitMilliseconds(3000))
-                .When(x => _steps.WhenIGetUrlOnTheApiGateway("/"))
-                .Then(x => _steps.ThenTheStatusCodeShouldBe(HttpStatusCode.OK))
-                .And(x => _steps.ThenTheResponseBodyShouldBe("Hello from Laura"))
-                .BDDfy();
-        }
-
-        private static void GivenIWaitMilliseconds(int ms)
-        {
-            Thread.Sleep(ms);
-        }
-
-        private void GivenThereIsAPossiblyBrokenServiceRunningOn(string url, string responseBody)
-        {
-            _serviceHandler.GivenThereIsAServiceRunningOn(url, async context =>
-            {
-                //circuit starts closed
-                if (_requestCount == 0)
-                {
-                    _requestCount++;
-                    context.Response.StatusCode = 200;
-                    await context.Response.WriteAsync(responseBody);
-                    return;
-                }
-
-                //request one times out and polly throws exception, circuit opens
-                if (_requestCount == 1)
-                {
-                    _requestCount++;
-                    await Task.Delay(1000);
-                    context.Response.StatusCode = 200;
-                    return;
-                }
-
-                //after break closes we return 200 OK
-                if (_requestCount == 2)
-                {
-                    context.Response.StatusCode = 200;
-                    await context.Response.WriteAsync(responseBody);
-                }
-            });
-        }
-
-        private void GivenThereIsAServiceRunningOn(string url, int statusCode, string responseBody, int timeout)
-        {
-            _serviceHandler.GivenThereIsAServiceRunningOn(url, async context =>
-            {
-                Thread.Sleep(timeout);
-                context.Response.StatusCode = statusCode;
-                await context.Response.WriteAsync(responseBody);
-            });
-        }
-
-        public void Dispose()
-        {
-            _serviceHandler?.Dispose();
-            _steps.Dispose();
-            GC.SuppressFinalize(this);
-        }
-    }
+
+namespace Ocelot.AcceptanceTests;
+
+public class PollyQoSTests : IDisposable
+{
+    private readonly Steps _steps;
+    private readonly ServiceHandler _serviceHandler;
+
+    public PollyQoSTests()
+    {
+        _serviceHandler = new ServiceHandler();
+        _steps = new Steps();
+    }
+
+    [Fact]
+    public void Should_not_timeout()
+    {
+        var port = PortFinder.GetRandomPort();
+
+        var configuration = new FileConfiguration
```

Comment:
Please, setup configuration being returned by private helper.
A lot of copy-pasted code!

Reply:
Ok, factories.

Reply:
@raman-m but that's not my code...

Reply:
Let's keep it like that for the minute?

Reply:
Let me help you...

Reply:
Mmmh, I have refactored it already. But it wasn't my code again..

Reply:
Just commit this https://github.com/ThreeMammals/Ocelot/pull/1753/commits/50f9b7a928da0c84e19eacf77dd258c0cc42a33f
I'm sorry, if we did this in parallel.